### PR TITLE
Ignore duplicated Orphan Vote mapped

### PR DESCRIPTION
**Pull request by thelazier**
This is just my quick hack to ignore duplicate activities when mapOrphanVotes.Insert did not really insert since it's possibly duplicated.

### CODE REVIEW

File: src/governance.cpp [hunk 1]
```diff
@@ -813,7 +815,9 @@ bool CGovernanceObject::ProcessVote(CNode* pfrom,
 {
     int nMNIndex = governance.GetMasternodeIndex(vote.GetVinMasternode());
     if(nMNIndex < 0) {
+        uint32_t nSize = mapOrphanVotes.GetSize();
```
Comment (UdjinM6) [inline src/governance.cpp]: uint32_t -> vote_mcache_t::size_type

File: src/governance.cpp [hunk 2]
```diff
@@ -619,7 +619,9 @@ bool CGovernanceManager::ProcessVote(CNode* pfrom, const CGovernanceVote& vote,
     uint256 nHashGovobj = vote.GetParentHash();
     object_m_it it = mapObjects.find(nHashGovobj);
     if(it == mapObjects.end()) {
+        size_type nSize = mapOrphanVotes.GetSize();
```
Comment (UdjinM6) [inline src/governance.cpp]: size_type -> vote_mcache_t::size_type

**Comment by thelazier** [comment]
I cannot find what is current nMaxSize so it may already be broken by my hack :)

**Comment by UdjinM6** [comment]
Not sure what you mean by
>I cannot find what is current nMaxSize

Isn't it just `mapOrphanVotes.GetMaxSize()` ?

**Comment by tgflynn** [comment]
I doubt this is the best way to deal with the problem of excessive log messages.

**Comment by UdjinM6** [comment]
It's not just about log messages, it's about processing same stuff (RequestGovernanceObject and AskForMn) many times if I get it right.

**Comment by tgflynn** [comment]
Yes, I still don't think this is the right way to fix that though.

**Comment by UdjinM6** [comment]
@tgflynn hmm, ok, any suggestions how to fix it in a better way or will you submit a better PR with the right fix?

**Comment by tgflynn** [comment]
I need to think about it some more, but I don't think we should merge this PR.

**Comment by thelazier** [comment]
where is it set, I don't see any code do SetMaxSize, only see number 0 in
header file..?

**Comment by UdjinM6** [comment]
@tgflynn well, that is not a very constructive response if you ask me ;) I would like to hear some arguments why this solution is not the right way to fix the issue or some suggestions how to improve it or to see some alternative solutions - any of these would be better than simply "because reasons" kind of response imo ;)

@thelazier hmm, true, so it's basically unlimited now and that's the edge case (GetMaxSize() == 0) of the edge case (GetMaxSize() == GetSize()) :D

**Comment by thelazier** [comment]
@UdjinM6 , I was blinded, `mapOrphanVotes` has max size initiated in `GovernanceManager` to be `MAX_CACHE_SIZE` ( `1000000` ).
In case `mapOrphanVotes` already at its max size, it will be `PruneLast()` in `Insert()` and possibly cause the `GetSize()` changed (`--nCurrentSize`) so it's not broken, IMO.

**Comment by tgflynn** [comment]
I don't see what the size of mapOrphanVotes has to do with whether the subsequent code should be run or not.

Also max size is 1000000 and I doubt there are that many votes on testnet, so I don't see how this would fix the problem.

**Comment by thelazier** [comment]
My assumption is that the subsequent code should only run once after OrphanVote got insertion properly. The duplicated votes being relayed later should be ignored. By the way, If that Orphan Vote got clean out by `CheckOrphanVotes` then it will be able to be inserted properly and the subsequent code will be run for that vote again.

**Comment by tgflynn** [comment]
That may be a good idea but I think it would be better to change CCacheMultimap::Insert to return a boolean indicating whether the insertion succeeded or not.

**Comment by thelazier** [comment]
Touching `CCacheMultimap::Insert` is beyond my assumption. The changes would impact to all `Insert()` calls to be revised.
Please feel free to make it better 😄

**Comment by UdjinM6** [comment]
@thelazier Ah, I missed that too :)

@tgflynn Yeah, I like the idea to change `Insert` signature, looking forward for a PR ;)

**Comment by UdjinM6** [comment]
I tested the patch and it feels like @thelazier nailed it - I see no more high cpu/bandwidth usage here. Good job 👍

**Comment by UdjinM6** [comment]
closing in fav of #1140

**Pull request closed without merging**